### PR TITLE
Adam/s3uploads

### PR DESCRIPTION
**Pull request by labseven**
Upload bobs to s3!

### CODE REVIEW

File: routes/s3api.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    .get(GETSignRequest)
+    .post(POSTS3Sign);
+
+  /*
+   * Respond to GET requests to /sign-s3.
+   * Upon request, return JSON containing the temporarily-signed S3 request and
+   * the anticipated URL of the image.
+   */
+  function GETSignRequest(req, res) {
+    console.log('GETSignRequest');
```
Comment (SeunginLyu) [inline routes/s3api.js]: remove log

File: routes/s3api.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    const fileType = req.query['file-type'];
+    const s3Params = {
+      Bucket: S3_BUCKET,
+      Key: fileName,
+      Expires: 60,
+      ContentType: fileType,
+      ACL: 'public-read'
+    };
+
+    console.log(s3Params);
```
Comment (SeunginLyu) [inline routes/s3api.js]: remove log

File: static/js/uploadfile.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
+SUBMIT_URL = null;
+
+/*
+  Function to carry out the actual PUT request to S3 using the signed request from the app.
+*/
+function uploadFile(file, signedRequest, url){
+  const xhr = new XMLHttpRequest();
+
+  console.log(url, signedRequest);
```
Comment (SeunginLyu) [inline static/js/uploadfile.js]: remove log

File: static/js/uploadfile.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+}
+
+/*
+ Bind listeners when the page loads.
+*/
+window.onload = function () {
+    document.getElementById('submit-button').onclick = submitBob;
+}
```
Comment (SeunginLyu) [inline static/js/uploadfile.js]: $('#submit-button').on("click touchstart", sumbmitBob)
lets use jQuery!

File: static/js/uploadfile.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+Dropzone.options.dropzoneInput = {
+  addedfile: function() {
+    // Remove old files
+    if(this.files.length > 1){
+      while(this.files.length > 1) { this.removeFile(this.files[0]); }
+    }
+    initUpload(this.files[0]);
+  },
+  drop: function() {
+    this.element.classList.remove("dz-drag-hover");
```
Comment (SeunginLyu) [inline static/js/uploadfile.js]: could be done with $(this).removeClass("dz-drag-hover")

**Review by SeunginLyu — COMMENTED** [review]
Looks good!

**Review by aidankmcl — APPROVED** [review]
Looks good, thanks for handling all of the SSL stuff!

**Pull request merged**